### ISSUE TITLE
Misaligned tiles when compiling on windows

### ISSUE THREAD

**Issue by Peeping-Peacock** [CleverRaven/Cataclysm-DDA#28505]
**Describe the bug**  
Compiling with MSYS2 or Visual studios leads to deformed and misaligned tiles on certain tilesets.

**To Reproduce**  
Compile cataclysm DDA on windows, use a tileset like thurztors, retrodays, or Tsu's. Some of the tile sets avoid this problem.

**Expected behavior**  
The experimentals look perfectly fine with identical graphical settings and versions.

**Screenshots**  
![experimental](https://user-images.githubusercontent.com/11481347/53820144-3700dd00-3f39-11e9-89bc-fc8a71cab919.png)

Looks fine, but then compiled-

![compiled](https://user-images.githubusercontent.com/11481347/53819511-0e2c1800-3f38-11e9-87ef-28b23128418e.png)

**Versions and configuration**
 - OS: Done on both windows 8.1 and windows 7
 - Game Version: 0.C-37575-gf07c447ac2

I originally thought this was a problem with me compiling the code but have since been told on discourse it affected others so I put up this issue. Could still be multiple people doing it wrong.

**Comment by ZhilkinSerg** [comment]
Can test after replacing `SDL_RenderSetScale` with `SDL_RenderSetIntegerScale` in `sdltiles.cpp`?

**Comment by Peeping-Peacock** [comment]
> 
> 
> Can test after replacing `SDL_RenderSetScale` with `SDL_RenderSetIntegerScale` in `sdltiles.cpp`?

After a find & replace all and recompiling, the tiles remain distorted.

**Comment by SomeDeadGuy** [comment]
I found the solution!
I am already implementing it in my tileset.
It is not an easy one. Basicly youll need to created tile for any possible side of the tile.
You will need separate graphics for 1 center, 4 corners, 4 t_connections, 2 edges, 4 end pieces, and 1 unconnected part. 16 separate sprites for each "thing" what can be connected to itself.
Only when this will be fixed.

**Comment by Peeping-Peacock** [comment]
> 
> 
> I found the solution!
> I am already implementing it in my tileset.
> It is not an easy one. Basicly youll need to created tile for any possible side of the tile.
> You will need separate graphics for 1 center, 4 corners, 4 t_connections, 2 edges, 4 end pieces, and 1 unconnected part. 16 separate sprites for each "thing" what can be connected to itself.
> Only when this will be fixed.

That sounds like a lot of work when the same outcome can apparently be accomplished by compiling on Linux, because all of the cataclysm experimental downloads lack this problem. Assuming the release builds for Windows are compiled on Linux, I have no idea who is in charge of the distribution bot and how they set it up.

**Comment by Peeping-Peacock** [comment]
> 
> 
> Can test after replacing `SDL_RenderSetScale` with `SDL_RenderSetIntegerScale` in `sdltiles.cpp`?

Turns out that works for visual studios but not MSYS2, I really should have tried both days ago, sorry about that and thank you so much for the help!

**Comment by Peeping-Peacock** [comment]
Unfortunately while this does fix the tiles problem with the visual studios method described in the "COMPILING.md" file, that compilation method is outdated and results in unplayable lag. So I've reopened this issue.

In fact, the whole "COMPILING.md" file is pretty much useless regarding windows compiling because nothing seems to work and more updated information for the same methods of compilation can be found under "COMPILING-MSYS.md" and "COMPILING-VS-VCPKG.md." Those two are more modern but the fix @ZhilkinSerg mentioned does not work for either, although everything else seems fine with no minute long input lag.

Considering how much of a pain this has been, I'd love to make a video to replace the 6 year out of date one on compiling cataclysm if anyone feels its worth their time to find a solution to this problem. I've spent more than a week trying to make this work but with zero experience on compiling I don't seem to be able to fix my own problem.
